### ISSUE TITLE
Panoramax: Adopt "copy id" feature from mapillary

### ISSUE THREAD

**Issue by gedankenstuecke** [openstreetmap/iD#10611]
### Description

**tl;dr: see [this comment below for good first issue description](https://github.com/openstreetmap/iD/issues/10611#issuecomment-2553846040)**

I've been loving the integration of the Panoramax street-level images in iD and started using it to add missing POI from the 360° images I recorded (c.f. this [changeset for an example](https://www.openstreetmap.org/changeset/160350919)). While doing so, I've tried to [add the `panoramax` key](https://wiki.openstreetmap.org/wiki/Key%3Apanoramax) for each new/updated POI using the corresponding image `UUID` to link back to the source images. 

### Agreed solution

* Take what was build in https://github.com/openstreetmap/id-tagging-schema/pull/1344
*  Allow it to work for the `panoramax` key
*  Abstract parts of that feature that might be shared between services in to shared files/functions
* Research if other imagery services have a key (share the wiki page); if so we should add the same feature in follow up PRs

**Comment by tordans** [comment]
@gedankenstuecke please test the feature we have for Mapillary from https://github.com/openstreetmap/iD/pull/10046 . I assume this would be a solution for your use case? In this case, we could create an issue to adapt this feature to Panoramax as well.

**Comment by gedankenstuecke** [comment]
@tordans I had a look at it, and that would be even more comfortable, at least for single images. It seems to not allow associating multiple images with the same object from what I can see?

**Comment by gedankenstuecke** [comment]
(Oh, and just to flag that adding the copy option would just be ~10 lines of JS and could be done quite quickly as a temporary solution!)

**Comment by tordans** [comment]
> It seems to not allow associating multiple images with the same object from what I can see?

That is correct. There was some discussion whether those imagery fields should default to allow multiple values in https://github.com/openstreetmap/id-tagging-schema/pull/1344 but as far as I remember we agreed to keep them optimized for one value.

**Comment by tordans** [comment]
> (Oh, and just to flag that adding the copy option would just be ~10 lines of JS and could be done quite quickly as a temporary solution!)

I think we should rather strive to only have one feature that does this but have it present for more imagery services.

**Comment by tordans** [comment]
### Good first issue description

- Take what was build in https://github.com/openstreetmap/id-tagging-schema/pull/1344
- Allow it to work for the `panoramax` key
- Abstract parts of that feature that might be shared between services in to shared files/functions
- Research if other imagery services have a key (share the wiki page); if so we should add the same feature in follow up PRs

Note that this does not have to wait for https://github.com/openstreetmap/id-tagging-schema/pull/1344; that PR will only improve the visibility of the tag as a preset field but we can already tag it now using the "Tags" section.

**Comment by gedankenstuecke** [comment]
> > It seems to not allow associating multiple images with the same object from what I can see?
> 
> That is correct. There was some discussion whether those imagery fields should default to allow multiple values in [openstreetmap/id-tagging-schema#1344](https://github.com/openstreetmap/id-tagging-schema/pull/1344) but as far as I remember we agreed to keep them optimized for one value.

Fair enough, I was thinking of the MapComplete case that was discussed in that PR too, which uses `panoramax:0:` and is used a bit already (c.f. [taginfo](https://taginfo.openstreetmap.org/keys/panoramax#similar)).

**Comment by tordans** [comment]
@gedankenstuecke could your rename this issue into "Panoramax: Adopt "copy id" feature from mapillary" (I cannot edit the title, unfortunately). This will make https://github.com/openstreetmap/iD/issues/10611#issuecomment-2553846040 clearer.

**Comment by gedankenstuecke** [comment]
done, and added a link to your good first issue comment in my initial/parent post!

**Comment by Vanshjain2701** [comment]
Can you please assign me this?

**Comment by tordans** [comment]
@Vanshjain2701 this is not how this repository handles issues and PRs. If you feel like you can tackle the ticket, go for it. It was pointed out to me that this is not the easiest "good first issue", so please only submit a PR that you feel confident about. Disclaimer: Please keep in mind that I have no official role in this project so your PR may or may not be accepted by Martin.

**Comment by AkashNegi1** [comment]
i've read the whole conversation and gone through each issue link provided and what i've got is that we've to implement 
![Screenshot 2025-01-07 203717](https://github.com/user-attachments/assets/ba481f63-0076-4b10-b471-8e1f70f39ce9)
this feature which was implemented in issue #10046 in the panoramax too 
![Screenshot 2025-01-07 203744](https://github.com/user-attachments/assets/35428966-1ed2-41ac-bc12-f35710cb76f5)
means in the above thing 
please correct me if I'm wrong anywhere
@tordans @gedankenstuecke

**Comment by tordans** [comment]
Yes, that is the goal. As well as abstracting what can/should be abstracted in the code and find a UI that works for all cases. The  screenshot shows that we need to find a different location for the button. An maybe icon due to the "+"  (zoom).

**Comment by avijitdas126** [comment]
Hello everyone,

I’ve implemented the "Copy ID" feature inspired by Mapillary as part of issue #10611. The current implementation adds a button/icon to easily copy the unique identifier of entities, improving the user experience.

I’d love feedback or suggestions to refine this feature further. If there are any specific improvements, alternative approaches, or edge cases you think we should consider, please feel free to share your thoughts.

Looking forward to collaborating with the community to make this feature as robust and useful as possible!

Thank you!

**Comment by gedankenstuecke** [comment]
Thanks @avijitdas126! Just to help others out: The PR in question is #10697!

**Comment by avijitdas126** [comment]
Thank you for pointing that out! I'll ensure to keep the context clear for others moving forward. Looking forward to any feedback on #10697!

**Comment by MohamedAli00949** [comment]
Hi @tordans ,

I have found the best way how to add the `copy id`  button to the Paranomax feature and solved this at this PR [#10856](https://github.com/openstreetmap/iD/pull/10856)

Also, I have completed the research about imagery services and I have found suitable services to which we can add this component it 
 I have completed the code changes at my local repo,
- [Bing street side](https://wiki.openstreetmap.org/wiki/Bing_Maps#Streetside_imagery)
- [Mapilio](https://wiki.openstreetmap.org/wiki/Mapilio)
- [KartaView](https://wiki.openstreetmap.org/wiki/KartaView)
- [Mapillary](https://wiki.openstreetmap.org/wiki/Mapillary)

**Comment by gedankenstuecke** [comment]
Thanks @MohamedAli00949 & @tyrasd for getting this done and over the finish line, it's much appreciated! <3
